### PR TITLE
LibJS: Do not throw a TypeError when sorted a detached TypedArray

### PR DESCRIPTION
**Pull request by trflynn89**
Normative change which fixes the following test262 test:
`test/built-ins/TypedArray/prototype/sort/sort-tonumber.js`

### CODE REVIEW

File: Userland/Libraries/LibJS/Contrib/Test262/$262Object.cpp [hunk 1]
```diff
@@ -62,17 +62,14 @@ JS_DEFINE_NATIVE_FUNCTION($262Object::create_realm)
...
     auto array_buffer = vm.argument(0);
     if (!array_buffer.is_object() || !is<ArrayBuffer>(array_buffer.as_object()))
         return vm.throw_completion<TypeError>(global_object);
+
     auto& array_buffer_object = static_cast<ArrayBuffer&>(array_buffer.as_object());
-    if (!same_value(array_buffer_object.detach_key(), vm.argument(1)))
-        return vm.throw_completion<TypeError>(global_object);
-    array_buffer_object.detach_buffer();
-    return js_null();
+    return JS::detach_array_buffer(global_object, array_buffer_object, vm.argument(1));
```
Comment (linusg) [inline Userland/Libraries/LibJS/Contrib/Test262/$262Object.cpp:72]: JS prefix not needed here, this is in LibJS land :)

File: Tests/LibJS/test-js.cpp [hunk 2]
```diff
@@ -76,6 +77,16 @@ TESTJS_GLOBAL_FUNCTION(mark_as_garbage, markAsGarbage)
     return JS::js_undefined();
 }
 
+TESTJS_GLOBAL_FUNCTION(detach_array_buffer, detachArrayBuffer)
+{
+    auto array_buffer = vm.argument(0);
+    if (!array_buffer.is_object() || !is<JS::ArrayBuffer>(array_buffer.as_object()))
+        return vm.throw_completion<JS::TypeError>(global_object);
```
Comment (linusg) [inline Tests/LibJS/test-js.cpp:84]:
```suggestion
        return vm.throw_completion<JS::TypeError>(global_object, ErrorType::NotAnObjectOfType, "ArrayBuffer");
```

File: Tests/LibJS/test-js.cpp [hunk 3]
```diff
@@ -76,6 +77,16 @@ TESTJS_GLOBAL_FUNCTION(mark_as_garbage, markAsGarbage)
...
+
+    auto& array_buffer_object = static_cast<JS::ArrayBuffer&>(array_buffer.as_object());
+    return JS::detach_array_buffer(global_object, array_buffer_object, vm.argument(1));
```
Comment (linusg) [inline Tests/LibJS/test-js.cpp:87]:
```suggestion
    auto& array_buffer = static_cast<JS::ArrayBuffer&>(array_buffer.as_object());
    return JS::detach_array_buffer(global_object, array_buffer, vm.argument(1));
```

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibJS/Contrib/Test262/$262Object.cpp (hunk 1 above)
Comment (trflynn89) [reply Userland/Libraries/LibJS/Contrib/Test262/$262Object.cpp:72]: We're in a class method also named `detach_array_buffer` so this is needed to fully qualify what we're actually calling

File: Tests/LibJS/test-js.cpp (hunk 3 above)
Comment (trflynn89) [reply Tests/LibJS/test-js.cpp:87]: We already have a value named `array_buffer` in this scope

Comment (trflynn89) [reply Tests/LibJS/test-js.cpp:87]: (The `array_buffer` / `array_buffer_object` names being copied from the $262Object impl)

Comment (linusg) [reply Tests/LibJS/test-js.cpp:87]: oh, I see. Not super nice, but fine :upside_down_face:

**Review by linusg — APPROVED** [review]

**Pull request merged**